bodhi/tests/server/test_notifications.py

```diff
@@ -201,99 +79,79 @@ def test_no_messages(self):
-            topic='demo.topic', msg={'new': 'package'})
-
-    def test_single_topic_many_messages(self, mock_fedmsg_publish):
-        """Assert many messages for a single topic are sent."""
-        session = Session()
-        notifications.publish('demo.topic', {'new': 'package'})
-        notifications.publish('demo.topic', {'newer': 'packager'})
-        session.commit()
-        self.assertEqual(2, mock_fedmsg_publish.call_count)
-        mock_fedmsg_publish.assert_any_call(
-            topic='demo.topic', msg={'new': 'package'})
-        mock_fedmsg_publish.assert_any_call(
-            topic='demo.topic', msg={'newer': 'packager'})
-
-    def test_multiple_topics(self, mock_fedmsg_publish):
-        """Assert messages with different topics are sent."""
-        session = Session()
-        notifications.publish('demo.topic', {'new': 'package'})
-        notifications.publish('other.topic', {'newer': 'packager'})
-        session.commit()
-        self.assertEqual(2, mock_fedmsg_publish.call_count)
-        mock_fedmsg_publish.assert_any_call(
-            topic='demo.topic', msg={'new': 'package'})
-        mock_fedmsg_publish.assert_any_call(
-            topic='other.topic', msg={'newer': 'packager'})
+            "An error occurred publishing %r after a database commit", message
+        )
+
+
+class FedMsgEncoderTests(base.BaseTestCase):
+    """Tests for the custom JSON encode ``FedMsgEncoder``."""
+
+    def test_default(self):
+        """Assert normal types are encoded the same way as the default encoder."""
+        self.assertEqual(
+            json.dumps('a string'),
+            json.dumps('a string', cls=notifications.FedMsgEncoder)
+        )
+
+    def test_default_sets_to_lists(self):
+        """Assert sets are converted to lists."""
+        self.assertEqual(
+            sorted(['a', 'set']),
+            sorted(notifications.FedMsgEncoder().default(set(['a', 'a', 'set'])))
+        )
+
+    def test_default_obj_with_json(self):
+        """Assert classes with a ``__json__`` function encode as the return of ``__json__``."""
+
+        class JsonClass(object):
+            def __json__(self):
+                return {'my': 'json'}
+
+        self.assertEqual(
+            {'my': 'json'},
+            notifications.FedMsgEncoder().default(JsonClass())
+        )
+
+    def test_default_datetime(self):
+        """Datetime objects should be converted to timestamps."""
+        timestamp = 1553016625
+        dt = datetime.datetime.fromtimestamp(timestamp)
+        self.assertEqual(
+            timestamp,
+            notifications.FedMsgEncoder().default(dt)
+        )
+
+    def test_default_time_struct(self):
+        """Time structures should be converted to timestamps."""
+        timestamp = 1553016625
+        dt = datetime.datetime.fromtimestamp(timestamp)
+        self.assertEqual(
+            timestamp,
+            notifications.FedMsgEncoder().default(dt.timetuple())
+        )
```

Comment:
Do we have any code that sends these time and set objects over messages?

Reply:
You're correct, we don't have any of those. I'll remove those branches.